src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -260,18 +262,75 @@ private void resetTuning() {
     numServesBlocking = 0;
   }
 
+  private static final Pattern RANGE = Pattern.compile("^bytes=(\\d+)-(\\d*)$");
+
   private void sendFile(RoutingContext context, String file, FileProps fileProps) {
     HttpServerRequest request = context.request();
+
+    // check if the client is making a range request
+    String range = request.getHeader("Range");
+    // parse it
+    Long offset = null;
+    // end byte is length - 1
+    Long end = fileProps.size() - 1;
```

Comment:
fileProps.size() is evaluated multiple times in the method. Better to evaluate it once.